lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1651,12 +1651,23 @@ emitAssociatedFunctionReference(ADContext &context, SILBuilder &builder,
         context.lookUpMinimalDifferentiationTask(originalFn, desiredIndices);
     if (!task) {
       if (originalFn->isExternalDeclaration()) {
-        context.emitNondifferentiabilityError(original, parentTask,
-            diag::autodiff_external_nondifferentiable_function);
-        return None;
+        llvm::StringRef module_name =
```

Comment:
Could you refactor this as a static function out of `emitAssociatedFunctionReference`? Say, `static bool isInREPL(SILModule &module)`

Reply:
Done.